docs/docs/guides/03-typescript.md

```diff
@@ -2,4 +2,226 @@
 title: Typescript
 ---
 
-🚧 coming soon 🚧
+# Development Guidelines
+
+This document provides guidelines for developing on the Strapi codebase using TypeScript. It covers various topics, including types location, generics, breaking changes, factories and more.
+
+## 1. Type Location
+
+When working with types on the Strapi codebase, it is essential to consider the location of types. These are the three common approaches and situations we document:
+
+### a. Colocation
+
+In the colocation approach, type definitions are colocated with the corresponding code. This means that types are defined in the same file or module where they are used.
+
+**This approach is suitable when the types are specific to a particular module**.
+
+_Note: In some situation, colocated types can be exported too if needed by other modules_
+
+Example:
+
+```typescript
+// config.ts
+interface Config {
+  prop1: string;
+  prop2: number;
+}
+
+function update(config: Config): Config {
+  // ...
+}
+```
+
+### b. Definitions Files
+
+**When types are shared across multiple files or modules, or when dealing with types that does not belong to a specific module**, it is recommended to use definition files (`.d.ts`) to separate type declarations from their usage.
```

Comment:
Does this make sense in the context of frontend components? i.e. we may export props from one components to be used to create the props of another, it feels strange that these props wouldn't live with it's component?

Reply:
I think in the context of components it definitely make sense to keep the props colocated with the component